### PR TITLE
LiveView Technical Support TableProviders

### PR DESCRIPTION
**Pull request by JimCAtSB**
This is a aggregated project with a java project as an artifact and a very simple LV project that uses the artifact.

There are no 3rd party libs. There are several java files that implement the LV public TableProvider API.

### CODE REVIEW

File: components/lv-ts-tableprovider/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# LiveView Technical Support TableProvider
+
+LiveView has a public TableProvider API that allows user to add
+*foreign* backend data sources. Such data sources are queried using
+whatever native language the backend implements. In the Technical
+Support TableProvider we have define a very simple "query language"
```
Comment (guillonapa) [inline components/lv-ts-tableprovider/README.md]: define -> defined ?

File: components/lv-ts-tableprovider/pom.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
+                      https://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+ 
+  <groupId>com.tibco.community</groupId>
+  <artifactId>ts-table-demo</artifactId>
+  <version>1.0</version>
+  <packaging>pom</packaging>
+
```
Comment (guillonapa) [inline components/lv-ts-tableprovider/pom.xml]: This is the place where the name and description for "this" sample should exist (what will be read by Studio to show the name and description)

File: components/lv-ts-tableprovider/LVSimple/src/main/resources/ts-client [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Copyright 2012-2018 TIBCO Software Inc.
```
Comment (guillonapa) [inline components/lv-ts-tableprovider/LVSimple/src/main/resources/ts-client]: Why is copyright only until 2018?

File: components/lv-ts-tableprovider/LVSimple/src/test/resources/FileSystemRead-test.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Simple tests for FileSystemRead Table
+
```
Comment (guillonapa) [inline components/lv-ts-tableprovider/LVSimple/src/test/resources/FileSystemRead-test.sh]: Shouldn't all script files in the resources dir have a copyright notice as in "ts-client"?

File: components/lv-ts-tableprovider/LVSimple/pom.xml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	<licenses>
+	    <license>
+	        <name>BSD 3-Clause License</name>
+	        <url>https://raw.githubusercontent.com/TIBCOSoftware/tibco-streaming-community/master/docs/Components-LICENSE</url>
+	        <distribution>repo</distribution>
+	    </license>
+	</licenses>
+ 
+	<properties>
+	    <com.tibco.ep.sb.studio.sample>true</com.tibco.ep.sb.studio.sample>
```
Comment (guillonapa) [inline components/lv-ts-tableprovider/LVSimple/pom.xml]: This pom shouldn't have this properties block

File: components/lv-ts-tableprovider/LVSimple/pom.xml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    <version>0.0.1-SNAPSHOT</version>
+    
+    <name>LiveView sample that uses TS TableProviders</name>
+    <description>LiveView TableProvider that allows LiveViewClients to get system logs, snapshots, system table contents, etc.</description>
+
+    <!-- common definitions for this version of StreamBase -->
+    <parent>
+        <groupId>com.tibco.ep.sb.parent</groupId>
+        <artifactId>ep-liveview-fragment</artifactId>
+        <version>10.5.1-SNAPSHOT</version>
```
Comment (guillonapa) [inline components/lv-ts-tableprovider/LVSimple/pom.xml]: missing a <relativePath/> inide the parent

File: components/lv-ts-tableprovider/TSTableProviders/pom.xml [hunk 7]
```diff
@@ -0,0 +1,36 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.tibco.ep.ldm.tstables</groupId>
+  <version>0.0.1-SNAPSHOT</version>
+  <name>LV technical support tables</name>
+  <description>TableProvider classes to add TS of LiveView</description>
+  
+    <properties>
+    	<com.tibco.ep.sb.studio.sample>true</com.tibco.ep.sb.studio.sample>
```
Comment (guillonapa) [inline components/lv-ts-tableprovider/TSTableProviders/pom.xml]: the com.tibco.ep.studio.sample property should not be defined here

**Comment by guillonapa** [comment]
The sample must also be declared as a module in the components maven aggregator (the pom)

File: components/lv-ts-tableprovider/pom.xml (hunk 2 above)
Comment (JimCAtSB) [reply components/lv-ts-tableprovider/pom.xml]: Ah, I put name/description in the two sub-projects. I should remove both those and put one at the top, yes?

File: components/lv-ts-tableprovider/LVSimple/src/main/resources/ts-client (hunk 3 above)
Comment (JimCAtSB) [reply components/lv-ts-tableprovider/LVSimple/src/main/resources/ts-client]: copy/paste error. changed to 2020!

File: components/lv-ts-tableprovider/LVSimple/src/test/resources/FileSystemRead-test.sh (hunk 4 above)
Comment (JimCAtSB) [reply components/lv-ts-tableprovider/LVSimple/src/test/resources/FileSystemRead-test.sh]: I guess so -  I added a copyright to these test scripts.

File: components/lv-ts-tableprovider/TSTableProviders/pom.xml (hunk 7 above)
Comment (JimCAtSB) [reply components/lv-ts-tableprovider/TSTableProviders/pom.xml]: OK, I removed from both sub-projects.

File: components/lv-ts-tableprovider/LVSimple/pom.xml (hunk 6 above)
Comment (JimCAtSB) [reply components/lv-ts-tableprovider/LVSimple/pom.xml]: Sorry, I haven't done (correctly anyway!) a project with modules. The sub project should look like this?

    <parent>
        <groupId>com.tibco.ep.sb.parent</groupId>
        <artifactId>ep-liveview-fragment</artifactId>
        <version>10.5.1-SNAPSHOT</version>
        <relativePath>../pom</relativePath>
    </parent>

**Pull request merged**